lua/hbac/setup.lua

```diff
@@ -1,18 +1,24 @@
-local command = require("hbac.command")
-local state   = require("hbac.state")
+local state = require("hbac.state")
 
-local M = {}
+local M = {
+	opts = {
+		threshold = 10,
+		autoclose = true,
+		close_buffers_with_windows = false,
+		close_command = function(bufnr)
+			vim.api.nvim_buf_delete(bufnr, {})
+		end,
+	},
+}
 
 local id = vim.api.nvim_create_augroup("hbac", {
-	clear = false
+	clear = false,
 })
 
 M.setup = function(opts)
-	opts = vim.tbl_extend("force", {
-		autoclose = true,
-		threshold = 10,
-	}, opts or {})
-
+	local command = require("hbac.command")
+	M.opts = vim.tbl_extend("force", M.opts, opts or {})
+	opts = M.opts
```

Comment:
At this point we populated the M.opts will all correct values, so we should use that
```suggestion
```